### ISSUE TITLE
ansible_os_name missing from linux facts

### ISSUE THREAD

**Issue by ansibot** [ansible/ansible#29731]
From @AseemKumar on 2016-07-27T09:10:52Z

##### ISSUE TYPE
- Feature Idea
##### COMPONENT NAME

setup
##### ANSIBLE VERSION

```
ansible 2.1.0.0
```
##### CONFIGURATION

default config.
##### OS / ENVIRONMENT

ansible is running on Ubuntu, managing windows 2008 + and Oracle linux systems.
##### SUMMARY

Comparing the facts returned by setup module, as I am trying to add the info to a DB and finding that a lot of fact names are not common.
Querying a windows box - "ansible_os_name": "Microsoft Windows Server 2008 R2 Standard"
This does not show up when a linux box is queried.
##### STEPS TO REPRODUCE

Run the setup module against a windows and linux box.

```
ansible host -i hosts.file -m setup
```

##### EXPECTED RESULTS

For a linux box, ansible_os_name can return full name like Trusty or Red Hat Enterprise Linux Server release 6.3 (Santiago)

on Ubuntu the codename from lsb_release -a can be used and on Redhat cat /etc/redhat-release
##### ACTUAL RESULTS

On linux box the fact did not show up.

Copied from original issue: ansible/ansible-modules-core#4260

**Comment by ansibot** [comment]
From @ansibot on 2016-07-27T09:10:52Z



@grossws ping, this issue is waiting for your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by ansibot** [comment]
From @robinro on 2016-07-27T09:10:52Z

I'm not convinced it is a good idea to add another linux naming fact. We already have `ansible_distribution*` and `ansible_lsb`, which have some redundancy and should contain all information you need.
One of the main issues with those two is that the information contained depends a lot on the distribution and I'm sure adding one more possibility will only increase the confusion.

**Comment by ansibot** [comment]
From @AseemKumar on 2016-07-27T09:10:52Z

I am unable to find ansible_lsb in the returned facts. My point is that if there can be as many common facts as possible, it will make life easier. Or maybe what is showing up in ansible_os_name on windows can be put in ansible_distribution so there is commonality? ansible_distribution and ansible_distribution_version return the same information, so ansible_distribution can be changed.

Let's say I am doing something with the facts returned from a mixed environment, then will have to probably create another common fact, and put what ansible_os_name returns on Windows and ansible_lsb returns on Linux in it.

**Comment by ansibot** [comment]
From @grossws on 2016-07-27T09:10:52Z

In my case `ansible_distribution` and `ansible_distribution_version` hold different values:

``` json
        "ansible_distribution": "CentOS", 
        "ansible_distribution_major_version": "7", 
        "ansible_distribution_release": "Core", 
        "ansible_distribution_version": "7.2.1511",
```

And `/etc/centos-release` contains `CentOS Linux release 7.2.1511 (Core)` (for same host).

As a workaround you can use something like

``` yaml
- set_fact: ansible_os="{{ ansible_os_name | d([ansible_distribution, ansible_system, 'release', ansible_distribution_version, '(' + ansible_distribution_release + ')'] | join(' ')) }}"
```

**Comment by ansibot** [comment]
cc @david_obrien
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
From @AseemKumar on 2016-07-27T09:10:52Z

Hi @grossws , I am gathering the facts by executing the adhoc command and storing all of it in a folder and then running other code on it to extract required values. How and where do I use the workaround that you have suggested?

**Comment by ansibot** [comment]
From @ansibot on 2016-07-27T09:10:52Z

@ansible, ping. This issue is waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by Akasurde** [comment]
@AseemKumar Can you please check if this still a problem for you and let us know ? thanks.

needs_info

**Comment by ansibot** [comment]
@AseemKumar This issue is waiting for your response. Please respond or the issue will be closed.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@AseemKumar You have not responded to information requests in this issue so we will assume it no longer affects you. If you are still interested in this, please create a new issue with the requested information.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)
